src/taco-tests-utils/projectHelper.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ *******************************************************
+ *                                                     *
+ *   Copyright (C) Microsoft. All rights reserved.     *
+ *                                                     *
+ *******************************************************
+ */
+
+"use strict";
+
+/* tslint:disable:no-var-requires */
+// var require needed for should module to work correctly
+// Note not import: We don't want to refer to shouldModule, but we need the require to occur since it modifies the prototype of Object.
+var shouldModule: any = require("should");
+/* tslint:enable:no-var-requires */
+
+import child_process = require ("child_process");
+import fs = require ("fs");
+import os = require ("os");
+import path = require ("path");
+import Q = require ("q");
+
+module TacoTestsUtils {
+    interface IKeyValuePair<T> {
+        [key: string]: T;
+    }
+
+    export class ProjectHelper {
+        public static checkPlatformVersions(platformsExpected: IKeyValuePair<string>, projectPath: string): Q.Promise<any> {
+            var platformsInstalled: string[] = ProjectHelper.getInstalledPlatforms(platformsExpected, projectPath);
+            var onWindows: boolean = process.platform === "win32";
+            var deferred: Q.Deferred<any> = Q.defer<any>();
+            return Q.all(platformsInstalled.map(function(platform: string): Q.Promise<any> {
+                var cmdName: string = "version";
+                if (onWindows) {
+                    cmdName = cmdName + ".bat";
+                }
+
+                var cmdPath: string = path.join(projectPath, "platforms", platform, "cordova", cmdName);
+                var versionProc: child_process.ChildProcess = child_process.spawn(cmdPath);
+                versionProc.stdout.on("data", function(data: any): void {
+                    var version: string = data.toString();
+                    if (!path) {
+                        deferred.reject("Verson string not found");
+                    } else {
+                        version.trim().should.be.equal(platformsExpected[platform]);
```

Comment:
This check should probably not be an assertion, or it should be in a try catch, since right now I think this will be an uncaught exception